.github/actions/draft-release/entrypoint.sh

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+
+set -euxo pipefail
+
+# Set up a git user
+git config user.name "release[bot]"
+git config user.email "actions@users.noreply.github.com"
+
+# Find last server-YYYY-MM-DD tag
+git fetch --unshallow --tags
+LAST_TAG=$(git tag | grep server | tail -n 1)
+
+# Find the marker in CHANGELOG.md
+INSERT_POINT=$(grep -n "^\-\-\-$" CHANGELOG.md | cut -f1 -d:)
+INSERT_POINT=$((INSERT_POINT+1))
+
+# Generate a release name
+RELEASE_NAME="server-$(date --rfc-3339=date)"
+
+# Assemble changelog entry
+rm -f temp-changes.txt
+touch temp-changes.txt
+{
+    echo "## $RELEASE_NAME"
+    echo ""
+    git log "$LAST_TAG"..HEAD --no-merges --oneline --pretty="format:- %s" --perl-regexp --author='^((?!dependabot).*)$'
```

Comment:
Our original proposal (and the one I wrote in #5866 ) was to grab all the PRs merged since the last release tag with the self-hosting label.
It turns out this is actually a bit of a fiddly thing to try and work out via the GH API (REST or GraphQL) as really the query we are trying to make is something like: "Find all pull requests where the commit we generated by squash-merging the PR is a child of commit X (where commit X is the last tagged commit)". It is certainly not impossible to work it out, but in the interests of keeping things fairly simple I'm going to propose we start off with this, which is every commit since the last release tag except for those where the author is dependabot. Starting with that and removing anything irrelevant at the manual review step seems like a good place to start and this has the advantage that it is easy to generate :) I suggest we try this and once we have manually reviewed a few we will have an idea of whether we want to revise it a bit or try and do something more sophisticated.